### PR TITLE
[OSD-23678] generate the console-ErrorBudgetBurn alerts only when the console is default route

### PR DESCRIPTION
**Pull request by bmeng**
Screenshot from 2024-07-30 11-02-40

As the metrics above shown, there is a built-in metric with name `console_url`, which will add a new entry everytime when the custom console route added/updated, and set the value to 1 only for the active console route. It can be used to identify if the console route is customized.

With above, append the following condition to the existing console-ErrorBudgetBurn alert

```
and count(console_url{url="$DEFAULT_CONSOLE_URL"} == 1) > 0
```
So it will alert SRE only when the active console_url is the default console route.

### CODE REVIEW

File: pkg/alert/prometheusrule.go [hunk 1]
```diff
@@ -117,6 +119,12 @@ func (r *multiWindowMultiBurnAlertRule) render(url string, percent string, names
 		" and " +
 		sufficientProbes(r.longWindow, labelSelector)
 
+	if namespacedName.Name == "console" {
+		defaultConsoleURL := strings.TrimSuffix(url, "/health")
```
Comment (xiaoyu74) [inline pkg/alert/prometheusrule.go:123]: How do we normally determine the console URL is default or custom? Just checking if it's in the `console` namespace?

**Comment by xiaoyu74** [comment]
With the current updated logic to only check the default active console URL, it looks that we are completely ignoring situations where requests fail to reach the custom console URL (e.g redirecting to a custom console URL and got certificate issues). Is that expected?

File: pkg/alert/prometheusrule.go (hunk 1 above)
Comment (bmeng) [reply pkg/alert/prometheusrule.go:123]: the `url` here comes from the `routeMonitor.Status.RouteURL` which was set to the default and I did not see it  would change

**Comment by bmeng** [comment]
> With the current updated logic to only check the default active console URL, it looks that we are completely ignoring situations where requests fail to reach the custom console URL (e.g redirecting to a custom console URL and got certificate issues). Is that expected?

yes, you can check the card for more details, but the goal is to monitor the console when it is not customized only

**Comment by xiaoyu74** [comment]
> > With the current updated logic to only check the default active console URL, it looks that we are completely ignoring situations where requests fail to reach the custom console URL (e.g redirecting to a custom console URL and got certificate issues). Is that expected?
> 
> yes, you can check the card for more details, but the goal is to monitor the console when it is not customized only

oh, I was thinking if we should have customer getting notifications or alerts when using custom console URL and got issues. Just checked the card, looks you've already discussed and that makes sense to me.
```
With that said, if the customer modified the console, that means the SLO does not make sense to us anymore, as it is not fully controlled by us.
If we feel the existing ingress operator or console operator cannot do monitoring them properly, we can enhance that part with OCP team.
```

File: pkg/alert/prometheusrule.go (hunk 1 above)
Comment (xiaoyu74) [reply pkg/alert/prometheusrule.go:123]: Sorry, I am still not quite clear how's the `url` comes from the `routeMonitor.Status.RouteURL` from the function calls. Do you mean it's derived from this struct?
```
// RouteMonitorStatus defines the observed state of RouteMonitor
type RouteMonitorStatus struct {
	// RouteURL is the url extracted from the Route resource
	RouteURL          string         `json:"routeURL,omitempty"`
	ServiceMonitorRef NamespacedName `json:"serviceMonitorRef,omitempty"`
	PrometheusRuleRef NamespacedName `json:"prometheusRuleRef,omitempty"`
	ErrorStatus       string         `json:"errorStatus,omitempty"`
}
```
But how's `RouteURL` above is determiend as the default URL? Is that explicitly defined something like `RouteURL: "https://console-openshift-console.apps.example.com/health`?

Comment (bmeng) [reply pkg/alert/prometheusrule.go:123]: The route fetched by the other function `GetRoute()` in routemonitor controller by reading the the route with name `console` in openshift-console ns, and append the suffix ("/health" here)

**Review by mjlshen — APPROVED** [review]
Not ideal, this is kind of an unexpected change from the "api" of route-monitor-operator, but it seems to be the simplest way to move forward.

/lgtm

**Pull request merged**